subprojects/scala/src/test/groovy/org/gradle/api/plugins/scala/ScalaPlugin3Test.groovy

```diff
@@ -22,30 +22,28 @@ import org.gradle.api.reporting.ReportingExtension
 import org.gradle.api.tasks.scala.ScalaDoc
 import org.gradle.test.fixtures.file.LeaksFileHandles
 import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.test.fixtures.Flaky
 import org.gradle.util.TestUtil
-import org.junit.Ignore
 import org.junit.Rule
 import org.junit.Test
-import spock.lang.Issue
+import org.junit.experimental.categories.Category
 
 import static org.gradle.api.tasks.TaskDependencyMatchers.dependsOn
 import static org.hamcrest.CoreMatchers.equalTo
 import static org.hamcrest.CoreMatchers.instanceOf
 import static org.hamcrest.MatcherAssert.assertThat
 
-@Ignore
-@Issue("https://github.com/gradle/gradle-private/issues/3440")
+@Category(Flaky)
```

Comment:
So we need to use `@Category` here, because this is a JUnit 3 test? Can't we add `@Flaky` here?

Reply:
Why this is a JUnit3? This is a JUni4 test. Jupiter engine can recognize JUnit4 `@Category` but can't recognize JUnit5 `@Tag`.

Reply:
Ok, so how can we make sure that we still have a link to an issue even for JUnit4 tests, since the link would now be lost.

Another option would be to convert this test to Spock, though I am not sure we'd want to do this with other possibly flaky JUnit4 tests.